samples/drivers/flash_shell/Kconfig

```diff
@@ -1,9 +0,0 @@
-# Copyright (c) 2020, Linaro Ltd.
-# SPDX-License-Identifier: Apache-2.0
-
-source "Kconfig.zephyr"
-
-if ARM_MPU
-config MPU_ALLOW_FLASH_WRITE
-	def_bool y
-endif
```

Comment:
Issue with this is that this is only selected by the following:
```
SOC_FLASH_NRF
SOC_FLASH_CC13XX_CC26XX
SOC_FLASH_MCUX
SOC_FLASH_GECKO
SOC_FLASH_STM32
SOC_FLASH_SAM0
SOC_FLASH_SAM
SOC_FLASH_SMARTBOND
SOC_FLASH_GD32
```
I would hazard a guess that there's far more SoCs supported in zephyr than that, this sample will now suddenly fail for them if they do a write test

Reply:
Good catch.

I cannot figure out how it would fail.
When `write` is used do you think it will result in a runtime failure:
```
 uart:~$ flash write 0x1000 0x11111111
    Write internal ERROR!
```
or something else?

Could it be solved in the flash_shell module?
Maybe by adding a `depends on` in the Kconfig or only include the `write` functionality if `MPU_ALLOW_FLASH_WRITE` or something else.

Reply:
I would assume the MPU would fault, but it's possible it would just show a write failure error. Possibly if it did a `select X if X` statement

Reply:
Can you elaborate on the `select X if X`?

I think this issue is part of `FLASH_SHELL` and not just this sample(?):
https://github.com/zephyrproject-rtos/zephyr/blob/af953a1d0742186cf34491fa2705cc2572aebe2d/drivers/flash/Kconfig#L52

How about:
```
if FLASH_SHELL
config FLASH_SHELL_WRITE_COMMANDS
default y
depends MPU_ALLOW_FLASH_WRITE
```

and only compile `write` commands if `FLASH_SHELL_WRITE_COMMANDS` is enabled?

Reply:
Hi @nordicjm.
Any idea on how to continue?

Reply:
Thanks @nordicjm. I've added a commit with your suggestion.